src/test/java/core/basesyntax/service/RegistrationServiceImplTest.java

```diff
@@ -0,0 +1,168 @@
+        user.setPassword("validPassword");
+        user.setAge(17);
+        assertThrows(RegistrationException.class, () -> registrationService.register(user));
+    }
+
+    @Test
+    void register_userWithAge18_notOk() {
+        User user = new User();
+        user.setLogin("validLogin");
+        user.setPassword("validPassword");
+        user.setAge(18);
+        assertThrows(RegistrationException.class, () -> registrationService.register(user));
+    }
+
+    @Test
+    void register_userGreater18_notOk() {
+        User user = new User();
+        user.setLogin("validLogin");
+        user.setPassword("validPassword");
+        user.setAge(19);
+        assertThrows(RegistrationException.class, () -> registrationService.register(user));
+    }
+
+    @Test
+    void register_userNegativeAge_notOk() {
+        User user = new User();
+        user.setLogin("validLogin");
+        user.setPassword("validPassword");
+        user.setAge(-1);
+        assertThrows(RegistrationException.class, () -> registrationService.register(user));
+    }
+
+    @Test
+    void register_passwordTooShort_notOk() {
+        User user = new User();
+        user.setLogin("validLogin");
+        user.setPassword("abc");
+        user.setAge(20);
+        assertThrows(RegistrationException.class, () -> registrationService.register(user));
+    }
+
+    @Test
+    void register_passwordLength5_notOK() {
+        User user = new User();
+        user.setLogin("validLogin");
+        user.setPassword("abcef");
+        user.setAge(20);
+        assertThrows(RegistrationException.class, () -> registrationService.register(user));
+    }
+
+    @Test
+    void register_passwordLengthMinimum_notOk() {
+        User user = new User();
+        user.setLogin("validLogin");
+        user.setPassword("abcefg");
+        user.setAge(20);
+        assertThrows(RegistrationException.class, () -> registrationService.register(user));
+    }
+
+    @Test
+    void register_passwordLengthMoreMinimum_Ok() {
+        User user = new User();
+        user.setLogin("validLogin");
+        user.setPassword("abcefgh");
+        user.setAge(20);
+        User registeredUser = registrationService.register(user);
+        assertNotNull(registeredUser);
+    }
+
+    @Test
+    void register_shortLogin_notOk() {
+        User user = new User();
+        user.setLogin("abc");
+        user.setPassword("password");
+        user.setAge(20);
+        assertThrows(RegistrationException.class, () -> registrationService.register(user));
+    }
+
+    @Test
+    void register_duplicateLogin_notOk() {
+        User user1 = new User();
+        user1.setLogin("uniqueLogin");
+        user1.setPassword("password");
+        user1.setAge(20);
+        registrationService.register(user1);
+
+        User user2 = new User();
+        user2.setLogin("uniqueLogin");
+        user2.setPassword("password123");
+        user2.setAge(21);
+        registrationService.register(user2);
```

Comment:
You should use `assertThrows` to verify that the `RegistrationException` is thrown for duplicate login. The current logic registers the user without checking for exceptions.